src/manifest/manifest.go

```diff
@@ -238,18 +228,25 @@ func (m *Manifest) CreateOrGetRecord(key model.Key) (record model.Record) {
 		panic(fmt.Errorf("unexpected type \"%s\"", key))
 	}
 	m.TrackRecord(record)
-	return record
+	return record, false
 }
 
 func (m *Manifest) PersistRecord(record model.Record) {
-	record.State().SetPersisted()
 	m.TrackRecord(record)
+	m.Naming.Attach(record.Key(), record.RelativePath())
+	record.State().SetPersisted()
+
 	m.lock.Lock()
 	defer m.lock.Unlock()
 	m.changed = true
 }
 
 func (m *Manifest) TrackRecord(record model.Record) {
+	// Resolve parent path, if record has been loaded from manifest.json
+	if record.GetParentPath() == "" {
```

Comment:
Ak sa do manifestu pridava nejaky novy objekt a nema vygenerovanu `parentPath`, tak sa vygeneruje.
Napr. nacitali sa konfiguracie z `manifest.json`, a teraz je potrebne vygenerovat `parentPath = branchPath`, kedze v `manifest.json` sa uklada iba `branchId` a potom sa to musi pospajat.